### ISSUE TITLE
[BUG] whitelist is too strong

### ISSUE THREAD

**Issue by FadeMind** [StevenBlack/hosts#57]
SITUATION

I whitelisting domains:
`copy.com`
`www.copy.com`

python script removed all entries with `copy.com` in name:

See: 
https://github.com/FadeMind/archbash/commit/2cd31bf13b8c3bc641400bfbf273c63b04e20af0
and after changes in hup code (and remove entries in `whitelist`
https://github.com/FadeMind/archbash/commit/96c1f8393738db22b262e5f6b62e12b7676cca5f
Instead of whitelisting, I using simple sed command, and now working as expected (see second commit)

@StevenBlack

**Comment by StevenBlack** [comment]
Good catch, Tomasz.

**Comment by omuravpn** [comment]
I discovered by accident that a trailing newline in the whitellist file, i.e:

```
# Domains added below will be ignored, For example:
# your-domain-name.com
icanhazip.com

```

is interpreted in such a way that it winds up whitelisting ALL hosts. The script returns "Success! Your new hosts file has been prepared. It contains 0 unique entries". Examining the newly generated hosts file, all the entries are commented out with a prepended `#`.

Oops. I think this is the same underlying issue FadeMind reported, so reporting as a comment instead of a new issue.

**Comment by StevenBlack** [comment]
Thanks @omuravpn!  Fixed in commit ffc17dd.
